doc/devdocs/locks.rst

```diff
@@ -0,0 +1,128 @@
+.. currentmodule:: Base
+
+****************************************************
+Proper maintenance and care of multi-threading locks
+****************************************************
+
+The following strategies are used to ensure that the code is dead-lock free
+(generally by addressing the 4th Coffman condition: circular wait).
```

Comment:
If you don't come from a formal CS background, you're also not likely to have learned these with a standard numbering.